### PR TITLE
Don't save irrelevant items into renewal configuration

### LINKED ISSUE #2503: Do not save irrelevant values to renewal conf files
Only things that would actually be read back should be written out.

### PR DESCRIPTION
**Pull request by schoen**
Fix #2503.

#2651 slightly limits the awesomeness of this fix.

I think the basic functionality is correct now. Not ready to merge yet because tests still need updating.

### CODE REVIEW

**Comment by schoen** [comment]
Also fixes #2277 because the logic here checks for both reasons not to save something into the renewal config.

**Comment by schoen** [comment]
Let's see how this does with Travis. I think I've now avoided making test coverage worse, although there are more meaningful tests that we could now add to directly confirm what does and doesn't get saved in renewal config files. I think those tests might be best added after #2651 is fixed.

File: letsencrypt/cli.py [hunk 1]
```diff
@@ -537,6 +537,20 @@ def diagnose_configurator_problem(cfg_type, requested, plugins):
     raise errors.PluginSelectionError(msg)
 
 
+def _relevant(option):
```
Comment (pde) [inline letsencrypt/cli.py]: Since it's being called from other modules, this function should not be private (ie not start with underscore).

**Comment by bmw** [comment]
#2669 should fix integration test failures

File: letsencrypt/storage.py [hunk 2]
```diff
@@ -127,6 +122,59 @@ def get_link_target(link):
...
 
+def _relevant(option):
+    """
+    Is this option one that could be restored for future renewal purposes?
+    :param str option: the name of the option
+
+    :rtype: bool
+    """
+    # The list() here produces a list of the plugin names as strings.
+    from letsencrypt import cli
```
Comment (SwartzCr) [inline letsencrypt/storage.py]: is it bad that these are imported in a helper function? Will that cause them to get imported multiple times per call to relevant_vaules or does python do some optimazation for this

**Comment by SwartzCr** [comment]
besides the one mentioned import issue this code looks good (and straightforward) to me

File: letsencrypt/storage.py (hunk 2 above)
Comment (nitmir) [inline letsencrypt/storage.py]: Python will import module only once independently of the number or `import module`. To reimport a module, one explicitly must call `reload(module)`.
But it is considered bad practice to not do import at the top of the file: https://www.python.org/dev/peps/pep-0008/#imports.

**Comment by bmw** [comment]
Merging with review from @SwartzCr.

**Pull request merged**